### DISCUSSION TITLE
initializing store w/ this.$state not updating state on store instance

### DISCUSSION THREAD

**Discussion by tomschulze** [vuejs/pinia#681] [Help and Questions] [answered]
hi,

not sure if i am just using bad practice code, so i opened a discussion instead of an issue.

I was using pinia 2.0.0-beta.2 before updating to rc9. with the update my store initialization pattern broke; actually it breaks from 2.0.0-beta.4 on.

i created two sandboxes with the same code but different pinia versions. beta2 works, rc9 doesn't. please checkout the `stores/cart.js` file and the `init()` function there.

basically i use `this.$state = data` to replace the whole store, however the getters return the default empty store state.

what am I doing wrong?

[codesandbox beta2](https://codesandbox.io/s/pinia-store-init-pattern-beta2-xu6yj)
[codesandbox rc9](https://codesandbox.io/s/pinia-store-init-pattern-rc9-gxskf)

**Comment by BenShelton** [comment, marked as answer]
This seems like a regression to me, the code comment for `$state` does say:
> State of the Store. Setting it will replace the whole state.

That evidently worked before but seems to break reactivity now if you do assign it.
My suggestion would be to create an issue for this.

However as a quick fix, you can instead use `$patch` which achieves the same result without breaking reactivity:

```ts
// change the following
this.$state = data
// to this
this.$patch(data)
```

[Forked Sandbox](https://codesandbox.io/s/pinia-store-init-pattern-rc9-forked-qopbv)

**Reply by tomschulze** [reply]
thanks for the workaround! i'll create an issue then.
